core/drivers/stm32_tamp.c

```diff
@@ -98,66 +110,294 @@
+	privcfgr = (privcfgr & _TAMP_PRIVCFGR_RIF_TAMP_PRIV ?
+		    _TAMP_PRIVCFG_TAMPPRIV : 0) |
+		   (privcfgr & _TAMP_PRIVCFGR_RIF_R1 ?
+		    (_TAMP_PRIVCFG_CNT1PRIV | _TAMP_PRIVCFG_BKPRWPRIV) : 0) |
+		   (privcfgr & _TAMP_PRIVCFGR_RIF_R2 ?
+		    (_TAMP_PRIVCFG_CNT2PRIV | _TAMP_PRIVCFG_BKPWPRIV) : 0);
+
+	/* Security and privilege RIF configuration */
+	io_clrsetbits32(base + _TAMP_PRIVCFGR,
+			_TAMP_PRIVCFGR_MASK & access_mask_sec_reg, privcfgr);
+	io_clrsetbits32(base + _TAMP_SECCFGR,
+			_TAMP_SECCFGR_BUT_BKP_MASK & access_mask_priv_reg,
+			seccfgr);
+
+	if (!stm32_tamp_dev->pdata.is_tdcid)
+		return;
+
+	for (i = 0; i < TAMP_RIF_RESOURCES; i++) {
+		if (!(BIT(i) & rif_conf->access_mask[0]))
+			continue;
+
+		io_clrsetbits32(base + _TAMP_CIDCFGR(i),
+				_TAMP_CIDCFGR_CONF_MASK,
+				rif_conf->cid_confs[i]);
+	}
+}
+
+static TEE_Result stm32_tamp_apply_bkpr_rif_conf(void)
+{
+	struct stm32_bkpregs_conf *bkpregs_conf =
+			stm32_tamp_dev->pdata.bkpregs_conf;
+	vaddr_t base = io_pa_or_va(&stm32_tamp_dev->pdata.base, 1);
+	int i = 0;
+
+	if (!bkpregs_conf)
+		panic("No backup register configuration");
+
+	for (i = 0; i < 4; i++) {
+		if (bkpregs_conf->rif_offsets[i] >
+		    (stm32_tamp_dev->hwconf1 & _TAMP_HWCFGR1_BKPREG))
+			return TEE_ERROR_NOT_SUPPORTED;
+	}
+
+	/* Fill the 3 TAMP_BKPRIFRx registers */
+	io_clrsetbits32(base + _TAMP_BKPRIFR(1), _TAMP_BKPRIFR_1_MASK,
+			bkpregs_conf->rif_offsets[0]);
+	io_clrsetbits32(base + _TAMP_BKPRIFR(2), _TAMP_BKPRIFR_2_MASK,
+			bkpregs_conf->rif_offsets[1]);
+	io_clrsetbits32(base + _TAMP_BKPRIFR(3), _TAMP_BKPRIFR_3_MASK,
+			bkpregs_conf->rif_offsets[2] |
+			SHIFT_U32(bkpregs_conf->rif_offsets[3],
+				  _TAMP_BKPRIFR_ZONE3_RIF2_SHIFT));
+
+	DMSG("Backup registers mapping :");
+	DMSG("********START of zone 1********");
+	DMSG("Protection Zone 1-RIF1 begins at register: 0");
+	DMSG("Protection Zone 1-RIF2 begins at register: %"PRIu32,
+	     bkpregs_conf->rif_offsets[0]);
+	DMSG("Protection Zone 1-RIF2 ends at register: %"PRIu32,
+	     bkpregs_conf->nb_zone1_regs ?
+	     bkpregs_conf->nb_zone1_regs - 1 : 0);
+	DMSG("********END of zone 1********");
+	DMSG("********START of zone 2********");
+	DMSG("Protection Zone 2-RIF1 begins at register: %"PRIu32,
+	     bkpregs_conf->nb_zone1_regs);
+	DMSG("Protection Zone 2-RIF2 begins at register: %"PRIu32,
+	     bkpregs_conf->rif_offsets[1]);
+	DMSG("Protection Zone 2-RIF2 ends at register: %"PRIu32,
+	     bkpregs_conf->rif_offsets[1] > bkpregs_conf->nb_zone1_regs ?
+	     bkpregs_conf->zone2_end - 1 : 0);
+	DMSG("********END of zone 2********");
+	DMSG("********START of zone 3********");
+	DMSG("Protection Zone 3-RIF1 begins at register: %"PRIu32,
+	     bkpregs_conf->zone2_end);
+	DMSG("Protection Zone 3-RIF0 begins at register: %"PRIu32,
+	     bkpregs_conf->rif_offsets[2]);
+	DMSG("Protection Zone 3-RIF2 begins at register: %"PRIu32,
+	     bkpregs_conf->rif_offsets[3]);
+	DMSG("Protection Zone 3-RIF2 ends at the last register: 128");
```

Comment:
replace `128` with `stm32_tamp_dev->hwconf1 & _TAMP_HWCFGR1_BKPREG` content, for consistency.